### ISSUE TITLE
Access to the path '/Portals/0/NBStore' is denied

### ISSUE THREAD

**Issue by jftsaraiva** [nbrightproject/NBrightBuy#60]
In a new installation, following DNN8 Basic Setup from http://nbsdocs.nbrightproject.org/Documentation/Downloads.aspx when I save my Admin > Setting  always get:

```
Access to the path '/Portals/0/NBStore' is denied.
```

The form data is saved but the message `Store Configuration needs to be done.` is always there and the directory /Portals/0/NBStore doesn't exist.

Even if I created NBStore Directory in 'File Management' - and this works fine - I'm getting the same error.

This is probably something with permissions, but I'm in Azure and don't know which one. Any suggestion? 

Full event log message:

```
AbsoluteURL:/Default.aspx
DefaultDataProvider:DotNetNuke.Data.SqlDataProvider, DotNetNuke
ExceptionGUID:d4c4203c-eef0-4d4a-b13e-055c3bb8022b
AssemblyVersion:8.0.4
PortalId:0
UserId:1
TabId:429
RawUrl:/Store
Referrer:http://xxxxxxxxxxx.azurewebsites.net/Store
UserAgent:Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.116 Safari/537.36
ExceptionHash:Wgm62z54aPCjkVwamQGLJQ==
Message:Access to the path '/Portals/0/NBStore' is denied.
StackTrace:
InnerMessage:Access to the path '/Portals/0/NBStore' is denied.
InnerStackTrace:
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.Directory.InternalCreateDirectory(String fullPath, String path, Object dirSecurityObj, Boolean checkHost)
   at System.IO.Directory.InternalCreateDirectoryHelper(String path, Boolean checkHost)
   at System.IO.Directory.CreateDirectory(String path)
   at Nevoweb.DNN.NBrightBuy.Admin.Settings.Update()
   at Nevoweb.DNN.NBrightBuy.Admin.Settings.CtrlItemCommand(Object source, RepeaterCommandEventArgs e)
   at System.Web.UI.WebControls.Repeater.OnItemCommand(RepeaterCommandEventArgs e)
   at System.Web.UI.WebControls.Repeater.OnBubbleEvent(Object sender, EventArgs e)
   at System.Web.UI.Control.RaiseBubbleEvent(Object source, EventArgs args)
   at System.Web.UI.WebControls.RepeaterItem.OnBubbleEvent(Object source, EventArgs e)
   at System.Web.UI.Control.RaiseBubbleEvent(Object source, EventArgs args)
   at System.Web.UI.WebControls.LinkButton.OnCommand(CommandEventArgs e)
   at System.Web.UI.WebControls.LinkButton.RaisePostBackEvent(String eventArgument)
   at System.Web.UI.WebControls.LinkButton.System.Web.UI.IPostBackEventHandler.RaisePostBackEvent(String eventArgument)
   at System.Web.UI.Page.RaisePostBackEvent(IPostBackEventHandler sourceControl, String eventArgument)
   at System.Web.UI.Page.RaisePostBackEvent(NameValueCollection postData)
   at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)
Source:
FileName:
FileLineNumber:0
FileColumnNumber:0
Method:
```

**Comment by hpkrekt** [comment]
I have the same problem and have also DNN installed on Azure. Any suggestion to solve this problem are welcomed.

**Comment by leedavi** [comment]
My guess is this is permissions issue.  I've not got much experience with hosting on Azure, but I don't think I'm doing anything special with file system access, so if DNN is working I'm not sure why NBS is not.  Check all your file system permissions.

**Comment by nicpitsch** [comment]
I have the same problem and error message, but on a VPS (Windows Server 2012). I double checked in IIS that NETWORK SERVICE has Modify permissions and that it is the Identity of the DefaultAppPool, which i use for the Site. DefaultAppPool is Integrated pipline mode and .NET CLR Version v4.0xxx. Also i can confirm that the Portal is a Site Type (and not a Child Portal) and that .NET Framework 4.6 is installed.

Another extension i use also creates subfolders on the portal with no problems. 

On my local machine i am able to setup NBStore according to the [DNN8 Basic Setup](http://nbsdocs.nbrightproject.org/LinkClick.aspx?fileticket=FhO3iTmUOv0%3d&tabid=466&portalid=6&mid=967&forcedownload=true) with no problem.

**Comment by leedavi** [comment]
Hi, this sounds very similar to the setup I use. So I would expect it to work!!

The only difference I have is that I usually just give full permission to the Network Service group.  You could try that and see if it works OK.

The only other possiblity I can think (which has happen to me before) is that for some reason when saving the store admin settings not all the correct folder got created.  Although I seem to recall a different error message for that.

But check what folders you have under the NBStore folder.  i would expect to see "docs","images","uploads" by default.  There should also be a NBSTemp folder under portal root.  But these folders shoudl match the names in the BO>Admin>Settings.

If they don't exist, try saving the setting again and see if they get created.

**Comment by leedavi** [comment]
NOTE: If you have issues with images not uploading correctly, you can use BO>Tools>Validate Store to ensure the settings are activated correctly.

**Comment by nicpitsch** [comment]
Hello, the folders don't get created when saving the Configuration. Therefore i already tried before to create them, also to register them with DNN File Management. No success.

I tried again with a fresh DNN8 installation:
- IIS8: New DefaultAppPool based Site with (now) full permission for Network Service.
- Install there DNN_Platform_8.0.4.226_Install.
- DNN: Install NBrightDNN_TemplateSys_v08.01.02.00.
- DNN: Install NBrightStore_03.05.01.00_DNN8_Install.
- DNN: Setup the pages and modules with nbs35-example-setup_DNN8.template. Check with DNN Page Management: everything seems ok. (I tried this earlier so far by hand).
- NBS: Go to NBS-Back-Office, set Store configuration then Save: Error.
- DNN: Create the folders by hand, NBS-Back-Office and Save: always the Same Error:

![2016-10-15_13-49-15](https://cloud.githubusercontent.com/assets/1652835/19409974/b3db8580-92e0-11e6-968c-6580bb28511e.jpg)

Event Viewer:
AbsoluteURL:/Default.aspx
DefaultDataProvider:DotNetNuke.Data.SqlDataProvider, DotNetNuke
ExceptionGUID:bc5fa6e8-188e-44a0-a145-3a7af5f4916c
AssemblyVersion:8.0.4
PortalId:0
UserId:1
TabId:83
RawUrl:/Admin/NBS-Back-Office
Referrer:http://xxxx.ch/Admin/NBS-Back-Office
UserAgent:Mozilla/5.0 (Windows NT 6.3; WOW64; rv:49.0) Gecko/20100101 Firefox/49.0
ExceptionHash:Khqk8/ul9SjY97rQwecqpw==
Message:Access to the path '/Portals/0/NBStore' is denied.
StackTrace:
InnerMessage:Access to the path '/Portals/0/NBStore' is denied.
InnerStackTrace:
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.Directory.InternalCreateDirectory(String fullPath, String path, Object dirSecurityObj, Boolean checkHost)
   at System.IO.Directory.InternalCreateDirectoryHelper(String path, Boolean checkHost)
   at Nevoweb.DNN.NBrightBuy.Admin.Settings.Update()
   at Nevoweb.DNN.NBrightBuy.Admin.Settings.CtrlItemCommand(Object source, RepeaterCommandEventArgs e)
   at System.Web.UI.WebControls.Repeater.OnItemCommand(RepeaterCommandEventArgs e)
   at System.Web.UI.WebControls.Repeater.OnBubbleEvent(Object sender, EventArgs e)
   at System.Web.UI.Control.RaiseBubbleEvent(Object source, EventArgs args)
   at System.Web.UI.WebControls.RepeaterItem.OnBubbleEvent(Object source, EventArgs e)
   at System.Web.UI.Control.RaiseBubbleEvent(Object source, EventArgs args)
   at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)
Source:
FileName:
FileLineNumber:0
FileColumnNumber:0
Method:
Server Name: xxxxx

**Comment by leedavi** [comment]
OK, can you create folders wirth the DNN file manger?  Or do you get an error there as well?

**Comment by nicpitsch** [comment]
I can create them with the file manager with no error.

**Comment by leedavi** [comment]
OK, I'm a little unsure what this could be!  I do recall, in the past, DNN having to do something special with the file system to overcome a security issue, but I'm not sure what and I'm unsure what it did to solve it.  I've never had a problem.

I'm using the standard .Net file operations for NBS, so I'm unsure why it's shoudln;t work if the permisisons are OK, which I assume they are if DNN can create folders.

I'll ask around and see if anyone can offer a reason!

**Comment by leedavi** [comment]
The only thing I can suggest as a idea would be to create another app pool dedictaed to this install and then applu full  permissions for the app pool onto the file system and see if that works.

**Comment by nicpitsch** [comment]
I did try with a separate app pool. No chance, allways that error. 

Why does this error stay also after creation of the NBS folders with DNN file manager?

**Comment by leedavi** [comment]
The error appears because although the folder exist, for some reason NBS doesn't have security rights to access them.  So I'm fairly sure it's a permissions issue, but from what you tell me of the setup I don't know what.

If you can allow me access to the server with RDP I could take a look to see if I can understand why,  Send me a PM with a login if you can.

**Comment by leedavi** [comment]
Just checking the DNN code and they are using the same System.IO.Directory class as me in NBS.

Have you checked that the permissions have cascaded to ALL the sub-folders correctly?

**Comment by nicpitsch** [comment]
Thanks. How respectively where can i send you a PM?

**Comment by leedavi** [comment]
I can;t find the PM system on GitHub either!

**Comment by leedavi** [comment]
This issue is fixed in v3.5.1.11, you can download the fix form DNN Store.
